src/common/guided_filter.c

```diff
@@ -498,36 +489,33 @@ static int _guided_filter_cl_impl(int devid,
 {
   const float eps = sqrt_eps * sqrt_eps; // this is the regularization parameter of the original papers
 
-  const int clwidth = ROUNDUPDWD(width, devid);
-  const int clheight = ROUNDUPDHT(height, devid);
-
-  cl_mem temp1 = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem temp2 = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem imgg_mean_r = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem imgg_mean_g = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem imgg_mean_b = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem img_mean = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem cov_imgg_img_r = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem cov_imgg_img_g = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem cov_imgg_img_b = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem var_imgg_rr = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem var_imgg_gg = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem var_imgg_bb = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem var_imgg_rg = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem var_imgg_rb = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem var_imgg_gb = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem a_r = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem a_g = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem a_b = dt_opencl_alloc_device(devid, clwidth, clheight, sizeof(float));
-  cl_mem b = temp2;
+  cl_mem temp1 = dt_opencl_alloc_device(devid, width, height, sizeof(float));
```

Comment:
For avoidance of doubt, `dt_opencl_alloc_device` does _not_ `ROUNDUPDWD`/`ROUNDUPDHT`. I don't know if that is needed here; they are sometimes (but seemingly not always) rounded before being passed to this function.

Reply:
Right. The size with rounding is not necessary here. I thought so while testing the Kahan sum and evaluating possible copy CL  image from and to host data.

I have been using such aligned buffer sizes for tiling stuff iirc.